modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -3787,4 +3762,51 @@ private AttributeInfo getAttributeInfo(AttributeInfoPool attrInfoPool, Attribute
     private boolean isWaitingOnNonBlockingAction() {
         return context.nonBlockingContext != null;
     }
+
+    private void calculateLength(int[] operands, StackFrame sf) {
+        int i = operands[0];
+        int cpIndex = operands[1];
+        int j = operands[2];
+
+        TypeRefCPEntry typeRefCPEntry = (TypeRefCPEntry) constPool[cpIndex];
```

Comment:
Since there are 4 different types, I would prefer a switch case statement. You can do that as an improvement or another PR

Reply:
Yah, I also thought about it. But then we have to do the null check for each case, for ref-types.. Now we do the null check for all ref-types only once.
WDYT?